framework/db/sqlite/QueryBuilder.php

```diff
@@ -394,6 +394,15 @@ public function build($query, $params = [])
             $sql = "$sql{$this->separator}$union";
         }
 
+        // replace aliases with table names
+        $sql = preg_replace_callback(
+            '/\\{\\{@([\w\-\. ]+)\\}\\}/',
+            function ($matches) use ($query) {
+                return '{{' . $query->getAlias($matches[1]) . '}}';
+            },
+            $sql
+        );
```

Comment:
it needs to be within Query and QueryBuilder layer because the lower layer classes like Schema and connection do not have anything to do with this feature.
About duplicated code problem in this case I have opened: https://github.com/yiisoft/yii2/issues/10815